scripts/queries.py

```diff
@@ -0,0 +1,108 @@
+queries = {
+    "content_model_distribution": """
+        SELECT ?model (COUNT(?obj) as ?count)
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/model#hasModel> ?model;
+        }
+        GROUP BY ?model
+    """,
+
+    "object_count": """
+        SELECT (COUNT(?obj) as ?count)
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/model#hasModel> <info:fedora/fedora-system:FedoraObject-3.0> .
+        }
+        LIMIT 1
+    """,
+
+    "active_deleted_count": """
+        SELECT (COUNT(?activeObj) AS ?ActiveCount) (COUNT(?deletedObj) AS ?DeletedCount)
+        FROM <#ri>
+        WHERE {
+            {
+                SELECT ?activeObj
+                WHERE {
+                    ?activeObj <info:fedora/fedora-system:def/model#state> <info:fedora/fedora-system:def/model#Active> .
+                }
+            } UNION {
+                SELECT ?deletedObj
+                WHERE {
+                    ?deletedObj <info:fedora/fedora-system:def/model#state> <info:fedora/fedora-system:def/model#Deleted> .
+                }
+            }
+        }
+    """,
+
+    "deleted_objects": """
+        SELECT ?obj
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/model#state> <info:fedora/fedora-system:def/model#Deleted>
+        }
+    """,
+
+    "datastream_distribution": """
+        SELECT ?datastream (COUNT(?datastream) as ?count)
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/model#hasModel> <info:fedora/fedora-system:FedoraObject-3.0>;
+            OPTIONAL {
+                ?obj <info:fedora/fedora-system:def/view#disseminates> ?c .
+                ?c <info:fedora/fedora-system:def/view#disseminationType> ?datastream ;
+            }
+        }
+        GROUP BY ?datastream
+    """,
+
+    "owner_distribution": """
+        SELECT ?owner (COUNT(?obj) as ?count)
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/model#ownerId> ?owner;
+        }
+        GROUP BY ?owner
+    """,
+
+    "collection_distribution": """
+        SELECT ?collection (COUNT(?obj) as ?count)
+        FROM <#ri>
+        WHERE {
+            ?obj <info:fedora/fedora-system:def/relations-external#isMemberOfCollection> ?collection;
```

Comment:
This could include collection objects that don't actually exist (and would become migration stub objects if they are still being referenced by objects when the migration gets run.)
A way to only get collections that exist would be:
```
            ?obj <info:fedora/fedora-system:def/relations-external#isMemberOfCollection> ?collection .
            ?collection <info:fedora/fedora-system:def/model#hasModel> <info:fedora/fedora-system:FedoraObject-3.0>
```
Testing for both of these scenarios and comparing is how we can check for migration stub objects (for completeness, we'd also need to do the same comparison using the isMemberOf, isConstituentOf, and any other relationships if there are others)

Reply:
Might make sense to add this query (with a different version of the query for each relationship, not just this `isMemberOfCollection` relationship:
```
SELECT ?collection (COUNT(?obj) as ?count)
FROM <#ri>
WHERE {
  ?obj <info:fedora/fedora-system:def/relations-external#isMemberOfCollection> ?collection
  FILTER NOT EXISTS {
    ?collection <info:fedora/fedora-system:def/model#hasModel> <info:fedora/fedora-system:FedoraObject-3.0>
  }
}
GROUP BY ?collection
```
This should give you a list of broken parent objects still referenced by child objects _and_ how many child objects have references to those broken parent objects.